hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestNSSummaryDiskUsageOrdering.java

```diff
@@ -0,0 +1,421 @@
+    return volumeId;
+  }
+
+  /**
+   * Create a bucket and add it to the Bucket Table.
+   * @return bucket Object ID
+   * @throws IOException
+   */
+  private long createBucket(String volumeName, String bucketName, long dataSize)
+      throws Exception {
+    String bucketKey =
+        reconOMMetadataManager.getBucketKey(volumeName, bucketName);
+    long bucketId = UUID.randomUUID().getMostSignificantBits() &
+        Long.MAX_VALUE; // Generate positive ID
+    OmBucketInfo bucketInfo = OmBucketInfo.newBuilder()
+        .setVolumeName(volumeName)
+        .setBucketName(bucketName)
+        .setObjectID(bucketId)
+        .setBucketLayout(getBucketLayout())
+        .setUsedBytes(dataSize)
+        .build();
+
+    reconOMMetadataManager.getBucketTable().put(bucketKey, bucketInfo);
+    return bucketId;
+  }
+
+  /**
+   * Create a directory and add it to the Directory Table.
+   * @return directory Object ID
+   * @throws IOException
+   */
+  private long createDirectory(long parentObjectId,
+                               long bucketObjectId,
+                               long volumeObjectId,
+                               String dirName) throws IOException {
+    long objectId = UUID.randomUUID().getMostSignificantBits() &
+        Long.MAX_VALUE; // Ensure positive ID
+    writeDirToOm(reconOMMetadataManager, objectId, parentObjectId,
+        bucketObjectId,
+        volumeObjectId, dirName);
+    return objectId;
+  }
+
+  /**
+   * Create a file and add it to the File Table.
+   * @return file Object ID
+   * @throws IOException
+   */
+  @SuppressWarnings("checkstyle:ParameterNumber")
+  private long createFile(String key,
+                          String bucket,
+                          String volume,
+                          String fileName,
+                          long parentObjectId,
+                          long bucketObjectId,
+                          long volumeObjectId,
+                          long dataSize) throws IOException {
+    long objectId = UUID.randomUUID().getMostSignificantBits() &
+        Long.MAX_VALUE; // Ensure positive ID
+    writeKeyToOm(reconOMMetadataManager, key, bucket, volume, fileName,
+        objectId,
+        parentObjectId, bucketObjectId, volumeObjectId, dataSize,
+        getBucketLayout());
+    return objectId;
+  }
+
+  private static ReconStorageContainerManagerFacade getMockReconSCM()
+      throws ContainerNotFoundException {
+    ReconStorageContainerManagerFacade reconSCM =
+        mock(ReconStorageContainerManagerFacade.class);
+    ContainerManager containerManager = mock(ContainerManager.class);
+
+    when(reconSCM.getContainerManager()).thenReturn(containerManager);
+    ReconNodeManager mockReconNodeManager = mock(ReconNodeManager.class);
+    when(reconSCM.getScmNodeManager()).thenReturn(mockReconNodeManager);
+    return reconSCM;
+  }
+
+  private static BucketLayout getBucketLayout() {
+    return BucketLayout.FILE_SYSTEM_OPTIMIZED;
+  }
```

Comment:
One last quick question: is this only tested with FSO buckets?

Reply:
Thanks for the comment @dombizita 

We've tested this manually for all bucket types. The changes don't really affect the result based on the bucket types, as the changes work directly towards the du response as you can see in the `VolumeEntityHandler` the `sortDiskUsageDescendingWithLimit()` is applied to the response list at the end when the entire response is created. so it does not really matter what the bucket type is.

Reply:
Thanks for explaining it @ArafatKhan2198!